python/cli/prelude_cli/views/detect.py

```diff
@@ -41,16 +44,19 @@ def register_endpoint(controller, host, serial_num, edr_id, tags):
 @click.argument('endpoint_id')
 @click.option('-h', '--host', help='hostname of this machine', type=str, default=None)
 @click.option('-e', '--edr_id', help='EDR id', type=str, default=None)
+@click.option('-p', '--partner', help='EDR partner',
+              type=click.Choice([c.name for c in Control] + ['None'], case_sensitive=False), default=Control.INVALID.name, show_default=False)
 @click.option('-t', '--tags', help='a comma-separated list of tags for this endpoint', type=str, default=None)
 @click.pass_obj
 @handle_api_error
-def update_endpoint(controller, endpoint_id, host, edr_id, tags):
+def update_endpoint(controller, endpoint_id, host, edr_id, partner, tags):
     """ Update an existing endpoint """
     with Spinner(description='Updating endpoint'):
         data = controller.update_endpoint(
             endpoint_id=endpoint_id,
             host=host,
             edr_id=edr_id,
+            partner_code=None if partner.upper() == 'NONE' else Control[partner.upper()].value,
```

Comment:
Seems odd to have the behavior on this differ on these 2 endpoints.

Reply:
Difference:
- register endpoint: 
    - we convert an invalid Control to None in the BE
    - there's no reason the user needs to set Control to "None" (that'll be the default)
- update endpoint:
     -  an invalid Control does not update the current value at all (consistent with other update fields where we silently ignore invalid fields)
     - user needs to be able to set Control to a None value if they want to remove the control